tests/cases/conformance/esnext/logicalAssignment/logicalAssignment2.ts

```diff
@@ -0,0 +1,29 @@
+// @strict: true
+// @target: esnext, es2020, es2015
+interface A {
+    foo: {
+        bar(): {
+            baz: 0 | 1 | 42 | undefined | ''
+        }
+        baz: 0 | 1 | 42 | undefined | ''
+    }
+    baz: 0 | 1 | 42 | undefined | ''
+}
+
+declare const result: A
+declare const a: A
+declare const b: A
+declare const c: A
+
+a.baz &&= result.baz
+b.baz ||= result.baz
+c.baz ??= result.baz
+
+a.foo["baz"] &&= result.foo.baz
+b.foo["baz"] &&= result.foo.baz
+c.foo["baz"] &&= result.foo.baz
```

Comment:
I think you meant for these to be the different operators:
```suggestion
b.foo["baz"] ||= result.foo.baz
c.foo["baz"] ??= result.foo.baz
```